phoenix-core/src/main/java/org/apache/phoenix/coprocessor/UngroupedAggregateRegionObserver.java

```diff
@@ -241,6 +243,16 @@ public void start(CoprocessorEnvironment e) throws IOException {
             e.getConfiguration().getInt(QueryServices.INDEX_REBUILD_RPC_RETRIES_COUNTER,
                 QueryServicesOptions.DEFAULT_INDEX_REBUILD_RPC_RETRIES_COUNTER));
         indexWriteProps = new ReadOnlyProps(indexWriteConfig.iterator());
+        final RegionCoprocessorEnvironment env = (RegionCoprocessorEnvironment) e;
+        String serverName = env.getRegionServerServices().getServerName().getServerName();
+        DelegateRegionCoprocessorEnvironment indexWriterEnv = new
+                DelegateRegionCoprocessorEnvironment(env, ConnectionType.INDEX_WRITER_CONNECTION);
+        // setup the global index writer
+        this.globalIndexRebuildWriter = new IndexWriter(indexWriterEnv, serverName + "-globalIndexRebuildWriter", false);
+    }
+
+    public IndexWriter getGlobalIndexRebuildWriter() {
```

Comment:
if other classes need access to this, better to pass it to them directly rather than pass an UARO and make them call this getter.

Reply:
We have to already pass the UARO coproc object directly. Are suggesting passing the writer directly too?